packages/types/src/primitive/Extrinsic/Extrinsic.ts

```diff
@@ -169,18 +179,53 @@ export default class Extrinsic extends Base<ExtrinsicV1> implements IExtrinsic,
     return this.raw.method;
   }
 
+  /**
+   * @description The nonce for this extrinsic
+   */
+  public get nonce (): NonceCompact {
+    return this.raw.signature.nonce;
+  }
+
   /**
    * @description The [[ExtrinsicSignature]]
    */
-  public get signature (): ExtrinsicSignature {
-    return this.raw.signature;
+  public get signature (): IHash {
+    return this.raw.signature.signature;
+  }
+
+  /**
+   * @description The [[Address]] that signed
+   */
+  public get signer (): Address {
+    return this.raw.signature.signer;
+  }
+
+  /**
+   * @description Forwards compat
+   */
+  public get tip (): BalanceCompact {
+    return this.raw.signature.tip;
+  }
+
+  /**
+   * @description Returns the encoded version flag
+  */
+  public get version (): number {
+    return this.versionFormat | (this.isSigned ? BIT_SIGNED : BIT_UNSIGNED);
+  }
+
+  /**
+   * @description Returns the raw version (not flagged)
+  */
+  public get versionFormat (): number {
```

Comment:
`versionFormat` is confusing, maybe `rawVersion` or `signedVersion`?

Reply:
Agree with the naming being sucky. The only alternative (as named elsewhere), that works is for me is actually `extrinsicVersion` (This was originally named that as well for consistency)

Reply:
Yeah, I prefer extrinsicVersion than versionFormat